Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -839,6 +839,71 @@ STATIC S3_SAVE_REG mS3SaveReg = {
   { { IO, WIDE32, { 0, 0}, (ACPI_BASE_ADDRESS + R_ACPI_IO_SMI_EN), 0x00000000 } }
 };
 
+VOID
+ProcessAllLocks (
+  IN VOID
+)
+{
+  SILICON_CFG_DATA  *SiliconCfgData;
+  UINT32            LpcBase;
+  UINTN             SpiBaseAddress;
+  UINT16            TcoBase;
+  UINTN             SpiBar0;
+
+  SiliconCfgData = (SILICON_CFG_DATA *)FindConfigDataByTag (CDATA_SILICON_TAG);
+  if ( (SiliconCfgData != NULL) && (SiliconCfgData->ECEnable == 1)){
+    //
+    // Enable decoding of I/O locations 62h and 66h to LPC
+    //
+    LpcBase = MmPciBase (0, PCI_DEVICE_NUMBER_PCH_LPC, 0);
+    MmioOr16 (LpcBase + R_LPC_CFG_IOE, B_LPC_CFG_IOE_ME1);
+
+    //
+    // Enable EC's ACPI mode to control power to motherboard during Sleep (S3)
+    //
+    IoWrite16 (EC_C_PORT, EC_C_ACPI_ENABLE);
+  }
+
+  // Lock down SPI for everything
+  SpiBaseAddress = GetDeviceAddr (OsBootDeviceSpi, 0);
+  SpiBaseAddress = TO_MM_PCI_ADDRESS (SpiBaseAddress);
+
+  // Set the BIOS Lock Enable and EISS bits
+  MmioOr8 (SpiBaseAddress + R_SPI_CFG_BC, (UINT8) (B_SPI_CFG_BC_LE | B_SPI_CFG_BC_EISS));
+
+  //
+  // Set Bios Interface Lock-Down (BILD)
+  // B0:D31:F05  Offset DCh Bit 07
+  //
+
+  // Set the BILD
+  MmioOr8 (SpiBaseAddress + R_SPI_CFG_BC, (UINT8)B_SPI_CFG_BC_BILD);
+
+  //
+  // set Flash Configuration Lock-Down (FLOCKDN)
+  // B0:D31:F05  Offset 04h Bit 15
+  //
+  SpiBar0 = (UINTN)PciRead32 (PCI_LIB_ADDRESS (0, PCI_DEVICE_NUMBER_PCH_SPI, PCI_FUNCTION_NUMBER_PCH_SPI, R_SPI_CFG_BAR0));
```

Comment:
SpiBaseAddress  was already got in line 869, don't need use SPI flash BDF again here.